libyul/backends/wasm/WasmCodeTransform.cpp

```diff
@@ -269,7 +268,13 @@ wasm::Expression WasmCodeTransform::operator()(ForLoop const& _for)
 	loop.statements += visit(_for.post.statements);
 	loop.statements.emplace_back(wasm::Branch{wasm::Label{loop.labelName}});
 
-	return { wasm::Block{breakLabel, make_vector<wasm::Expression>(move(loop))} };
+	if (_for.pre.statements.empty())
+		return wasm::Block{breakLabel, make_vector<wasm::Expression>(move(loop))};
+
+	std::vector<wasm::Expression> result;
+	result += visit(_for.pre.statements);
+	result += make_vector<wasm::Expression>(move(loop));
+	return wasm::Block{breakLabel, move(result)};
```

Comment:
Isn't this equivalent to
```suggestion
	return wasm::Block{breakLabel, visit(_for.pre.statements) + make_vector<wasm::Expression>(move(loop))};
```
?

But could you try to move the `visit(_for.pre.statemenst)` to the front again, so that it is visited before the loop? It should not make a difference, but I think it's better to be safe than sorry.

Reply:
Sadly `return wasm::Block{breakLabel, visit(_for.pre.statements) + make_vector<wasm::Expression>(move(loop))};` is not working because std::variant doesn't provide a copy constructor. I think then the previous solution was then nicer.

Reply:
Can you please still find a solution that
 - visits the pre statements before everything else
 - reduces code duplication

Reply:
yes, sure